app/src/main/java/com/codingblocks/cbonlineapp/workers/SectionService.kt

```diff
@@ -0,0 +1,201 @@
+package com.codingblocks.cbonlineapp.workers
+
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.app.Service
+import android.content.Context
+import android.content.Intent
+import android.graphics.BitmapFactory
+import android.os.Environment
+import android.os.IBinder
+import android.util.Log
+import androidx.core.app.NotificationCompat
+import androidx.core.content.ContextCompat
+import androidx.work.WorkManager
+import com.codingblocks.cbonlineapp.R
+import com.codingblocks.cbonlineapp.database.ContentDao
+import com.codingblocks.cbonlineapp.database.SectionWithContentsDao
+import com.codingblocks.cbonlineapp.database.models.SectionContentHolder
+import com.codingblocks.cbonlineapp.util.DOWNLOAD_CHANNEL_ID
+import com.codingblocks.cbonlineapp.util.RUN_ATTEMPT_ID
+import com.codingblocks.cbonlineapp.util.SECTION_ID
+import com.codingblocks.onlineapi.CBOnlineLib
+import com.google.gson.JsonObject
+import com.vdocipher.aegis.media.ErrorDescription
+import com.vdocipher.aegis.offline.DownloadOptions
+import com.vdocipher.aegis.offline.DownloadRequest
+import com.vdocipher.aegis.offline.DownloadSelections
+import com.vdocipher.aegis.offline.DownloadStatus
+import com.vdocipher.aegis.offline.OptionsDownloader
+import com.vdocipher.aegis.offline.VdoDownloadManager
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import org.koin.android.ext.android.inject
+import retrofit2.Response
+import java.io.File
+
+/**
+ * A Foreground Service to download files
+ */
+
+class SectionService : Service(), VdoDownloadManager.EventListener {
```

Comment:
@championswimmer can you take at this other than this most of things are removing build-time warnings and rearranging some stuff